events.go

```diff
@@ -216,41 +216,56 @@ var fakeCancelReason = []byte("internal: cancel")
 var fakeTimeoutReason = []byte("internal: call state timeout")
 var fake2xxReason = []byte("internal: implied OK")
 
-func (d *EventData) EncryptIP(key [16]byte) error {
-	var err error
-	if d.Src.To4() != nil && d.Dst.To4() != nil {
-		err = d.EncryptIPv4(key)
-	} else if d.Src.To16() != nil && d.Dst.To16() != nil {
-		err = d.EncryptIPv6(key)
-	} else {
-		err = errors.New("EncryptIP: broken IP address")
+// setIPv4Bytes sets the bytes of an IPv4 address preserving the format
+// (i.e. either plain IPv4 or IPv4 address represented as an IPv6 address)
+// it assumes "ip" is an IPv4 address
+func setIPv4Bytes(ip net.IP, b [4]byte) (err error) {
+	switch len(ip) {
+	case net.IPv4len:
+		ip = b[:]
```

Comment:
Note that here ip will be changed to _point_ to b (to the ip slice will be assigned a slice made out of the entire b[], this means the slice pointer will end up pointing to b). Since this is going to be used with the event, you should change the ip content and not the slice pointer (so either copy(ip, b[:]) or for ... ip[i] = b[i] ...).
In the IPv6 case you did copy.